program-runtime/src/execution_budget.rs

```diff
@@ -0,0 +1,278 @@
+            heap_size: u32::try_from(solana_program_entrypoint::HEAP_LENGTH).unwrap(),
+            heap_cost: DEFAULT_HEAP_COST,
+            mem_op_base_cost: 10,
+            alt_bn128_addition_cost: 334,
+            alt_bn128_multiplication_cost: 3_840,
+            alt_bn128_pairing_one_pair_cost_first: 36_364,
+            alt_bn128_pairing_one_pair_cost_other: 12_121,
+            big_modular_exponentiation_base_cost: 190,
+            big_modular_exponentiation_cost_divisor: 2,
+            poseidon_cost_coefficient_a: 61,
+            poseidon_cost_coefficient_c: 542,
+            get_remaining_compute_units_cost: 100,
+            alt_bn128_g1_compress: 30,
+            alt_bn128_g1_decompress: 398,
+            alt_bn128_g2_compress: 86,
+            alt_bn128_g2_decompress: 13610,
+        }
+    }
+
+    /// Returns cost of the Poseidon hash function for the given number of
+    /// inputs is determined by the following quadratic function:
+    ///
+    /// 61*n^2 + 542
+    ///
+    /// Which aproximates the results of benchmarks of light-posiedon
+    /// library[0]. These results assume 1 CU per 33 ns. Examples:
+    ///
+    /// * 1 input
+    ///   * light-poseidon benchmark: `18,303 / 33 ≈ 555`
+    ///   * function: `61*1^2 + 542 = 603`
+    /// * 2 inputs
+    ///   * light-poseidon benchmark: `25,866 / 33 ≈ 784`
+    ///   * function: `61*2^2 + 542 = 786`
+    /// * 3 inputs
+    ///   * light-poseidon benchmark: `37,549 / 33 ≈ 1,138`
+    ///   * function; `61*3^2 + 542 = 1091`
+    ///
+    /// [0] https://github.com/Lightprotocol/light-poseidon#performance
+    pub fn poseidon_cost(&self, nr_inputs: u64) -> Option<u64> {
+        let squared_inputs = nr_inputs.checked_pow(2)?;
+        let mul_result = self
+            .poseidon_cost_coefficient_a
+            .checked_mul(squared_inputs)?;
+        let final_result = mul_result.checked_add(self.poseidon_cost_coefficient_c)?;
+
+        Some(final_result)
+    }
+
+    pub fn apply_overrides(&self, overrides: &SVMTransactionBudgetOverrides) -> Self {
+        let mut budget = *self;
+        if let Some(compute_unit_limit) = overrides.compute_unit_limit {
+            budget.compute_unit_limit = compute_unit_limit;
+        }
+        if let Some(heap_size) = overrides.heap_size {
+            budget.heap_size = heap_size;
+        }
+        budget
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct SVMTransactionBudgetOverrides {
+    pub compute_unit_limit: Option<u64>,
+    pub heap_size: Option<u32>,
+}
+
+impl SVMTransactionBudgetOverrides {
+    pub fn no_overrides() -> Self {
+        Self {
+            compute_unit_limit: None,
+            heap_size: None,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct SVMTransactionComputeBudgetAndLimits {
+    pub budget_overrides: SVMTransactionBudgetOverrides,
+    pub loaded_accounts_bytes: NonZeroU32,
+    pub priority_fee: u64,
```

Comment:
Is it necessary to include `priority_fee`? I don't see the use case of it in this PR, maybe it is used elsewhere within SVM?

Reply:
It's being used in `transaction_processor.rs`, here: https://github.com/anza-xyz/agave/pull/5127/files#diff-9cdf3f551fba4b8b92dbdd58af9c878564da6720db0beffd6e64ec3f36de375aR608

The diff probably doesn't expand automatically due to amount of changes in file. You need to load the diff to see it.

Reply:
as discussed on discord - we will likely replace this soon with the total fee, and remove fee-calculation from `svm` processing.